pkg/logs/launchers/container/tailerfactory/defaults.go

```diff
@@ -21,7 +23,11 @@ import (
 // encounters along the way
 func (tf *factory) defaultSourceAndService(source *sources.LogSource, logWhat containersorpods.LogWhat) (sourceName, serviceName string) {
 	getContainer := func(containerID string) (*workloadmeta.Container, error) {
-		return tf.workloadmetaStore.GetContainer(containerID)
+		wmeta, ok := tf.workloadmetaStore.Get()
```

Comment:
Would it make more sense for the logs agent to depend on a non-optional workloadmeta? I don't think this code is ever intended to run without a valid workloadmeta instance.

IMO if possible optional should be reserved for code that is expected to work both with and without a given dependency. When I encountered this with the collector and ended up providing [both an optional and non-optional version](https://github.com/DataDog/datadog-agent/blob/main/comp/collector/collector/collectorimpl/collector.go#L78) which worked quite well - as there are a mix of components that can run without the collector, and some that require it.

Reply:
Understood. Thanks. Yeah build tags do complicate things, and also decrease visibility for issues like this :(